### PR TITLE
oai v1 embeddings

### LINKED ISSUE #12943: OpenAIEmbeddings() does not work because of these bugs
### System Info

Python Version: 3.11
LangChain Version: 0.0.331
OpenAI Version: 1.0.0

### Who can help?

@hwchase17, @agola11, @eyurtsev

### Information

- [ ] The official example notebooks/scripts
- [X] My own modified scripts

### Related Components

- [ ] LLMs/Chat Models
- [X] Embedding Models
- [ ] Prompts / Prompt Templates / Prompt Selectors
- [ ] Output Parsers
- [ ] Document Loaders
- [X] Vector Stores / Retrievers
- [ ] Memory
- [ ] Agents / Agent Executors
- [ ] Tools / Toolkits
- [ ] Chains
- [ ] Callbacks/Tracing
- [ ] Async

### Reproduction

The following error has been caused due to the recent change in version of OpenAI to 1.0.0
**Use OpenAI==0.28.1 to fix this error**
With the code:
`embeddings = OpenAIEmbeddings()`

The error produced is:
`AttributeError: module 'openai' has no attribute 'Embedding'. Did you mean: 'embeddings'?`

I went through the `langchain/embeddings/openai.py` file and then changed `value["client"] = openai.Embedding` to `value["client"] = openai.embeddings`, but then I receive this new error:

`AttributeError: module 'openai' has no attribute 'error'` in the same file (`langchain/embeddings/openai.py`)

### Expected behavior

There should be no error when calling this function.

### PR DESCRIPTION
**Pull request by efriis**
Initial PR to get OpenAIEmbeddings working with the new sdk

fyi @rlancemartin 

Fixes #12943

### CODE REVIEW

File: libs/langchain/langchain/embeddings/openai.py [hunk 1]
```diff
@@ -231,7 +250,7 @@ def build_extra(cls, values: Dict[str, Any]) -> Dict[str, Any]:
         values["model_kwargs"] = extra
         return values
 
-    @root_validator(pre=True)
```
Comment (baskaryan) [inline libs/langchain/langchain/embeddings/openai.py:234]: do we have to change

File: libs/langchain/langchain/embeddings/openai.py [hunk 2]
```diff
@@ -376,7 +415,10 @@ def _get_len_safe_embeddings(
                 input=tokens[i : i + _chunk_size],
                 **self._invocation_params,
             )
-            batched_embeddings.extend(r["embedding"] for r in response["data"])
+            if _is_openai_v1():
+                batched_embeddings.extend(r.embedding for r in response.data)
```
Comment (baskaryan) [inline libs/langchain/langchain/embeddings/openai.py:419]: you could also convert response to a dict upfront and keep rest of code same
```python
if not isinstance(response, dict):
  response = response.dict()
```

File: libs/langchain/langchain/embeddings/openai.py (hunk 1 above)
Comment (efriis) [reply libs/langchain/langchain/embeddings/openai.py:234]: The other oai types don't do pre, and having pre makes it fail on request_timeout iirc. Can jump through some hoops if we want to keep it

Comment (efriis) [reply libs/langchain/langchain/embeddings/openai.py]: changed

**Comment by WebsheetPlugin** [comment]
+1

**Pull request merged**